mypyc/lib-rt/str_ops.c

```diff
@@ -201,4 +199,40 @@ Py_ssize_t CPyStr_Size_size_t(PyObject *str) {
         return PyUnicode_GET_LENGTH(str);
     }
     return -1;
-}
\ No newline at end of file
+}
+
+PyObject* CPy_Decode(PyObject *obj) {
+    if (PyBytes_Check(obj)) {
+        return PyUnicode_Decode(((PyBytesObject *)obj)->ob_sval,
+                                ((PyVarObject *)obj)->ob_size,
+                                NULL, NULL);
+    } else {
+        return PyUnicode_FromEncodedObject(obj, NULL, NULL);
+    }
+}
+
+PyObject* CPy_DecodeWithEncoding(PyObject *obj, PyObject *encoding) {
+    const char *enc = PyUnicode_AsUTF8AndSize(encoding, NULL);
+    if (!enc) return NULL;
+    if (PyBytes_Check(obj)) {
+        return PyUnicode_Decode(((PyBytesObject *)obj)->ob_sval,
+                                ((PyVarObject *)obj)->ob_size,
+                                enc, NULL);
+    } else {
+        return PyUnicode_FromEncodedObject(obj, enc, NULL);
+    }
+}
+
+PyObject* CPy_DecodeWithErrors(PyObject *obj, PyObject *encoding, PyObject *errors) {
```

Comment:
As discussed offline, I'd prefer to have three primitive definitions that all call a single C function, i.e. we'd only duplicate the `method_op` declarations, not C implementations. Some primitives can provide fixed extra arguments, but without using the for loop for clarity. Sorry for the extra back and forth!